linalg/simd/x86.hpp

```diff
@@ -12,10 +12,14 @@
 #ifndef MFEM_SIMD_X86_HPP
 #define MFEM_SIMD_X86_HPP
 
+#if !(defined(__CUDACC__) || defined(__HIPCC__))
```

Comment:
This definitely should be removed -- at least with `nvcc` the x86 intrinsics work fine.